### PR TITLE
Add Config Init Command

### PR DESCRIPTION
**Pull request by mahirshah**
Add command to initialize a checkuprc file via `checkup config:init`.
Move to a true multi command cli, by moving main checkup command to the
`run` command

### CODE REVIEW

File: packages/core/src/configuration/config-init-service.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import CheckupConfigService from './checkup-config-service';
+import { CheckupConfig, CheckupConfigFormat, ConfigMapper } from '../types';
+import addPlugin from './mappers/add-plugin';
+import getInitializationConfigLoader from './loaders/get-initialization-loader';
+import CosmiconfigService from './cosmiconfig-service';
+
+type QuestionNames = typeof ConfigInitService.questions[number]['name'];
+type Answers = Record<QuestionNames, string>;
+
+export default class ConfigInitService {
```
Comment (scalvert) [inline packages/core/src/configuration/config-init-service.ts]: I would much prefer this class was included in a directory that implied its intent.

`packages/core/src/generators/config/init.ts`

This is feeling very java-y :)

If we keep all our generators in that structure, you can organize and find them very easily on disc.

```
// generators/config/init.ts
checkup config:init 
```

```
// generators/task.ts
checkup task <taskName>
```

```
// generators/plugin.ts
checkup plugin <pluginName>
```

Inside the `generators` directory we can have a `generators/templates` directory that points to the files on disc. I'd really like to just use yoman here to create and invoke templates, similar to what oclif does. It's very powerful, and comes with all the inquirer, etc stuff you need.

File: packages/core/src/configuration/mappers/add-plugin.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { ConfigMapper } from '../../types';
+
+const addPlugin: (pluginName: string) => ConfigMapper = (pluginName: string) => config => {
```
Comment (scalvert) [inline packages/core/src/configuration/mappers/add-plugin.ts]: Why does this need to be in an external module? Can it not be inlined in the generator?

File: packages/core/src/configuration/config-init-service.ts (hunk 1 above)
Comment (scalvert) [reply packages/core/src/configuration/config-init-service.ts]: Example: https://github.com/oclif/oclif/blob/master/src/generators/command.ts#L10

File: packages/cli/src/commands/config/init.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { Command } from '@oclif/command';
```
Comment (scalvert) [inline packages/cli/src/commands/config/init.ts]: Thoughts on making a top level command called `generate`, a la Ember?

```
checkup generate config
checkup generate task <taskName>
checkup generate plugin <pluginName>
```

Comment (mahirshah) [reply packages/cli/src/commands/config/init.ts]: hmm I think generate makes sense for `task` and `plugin`, but for config it seems like `config:init` or similar is better. Thoughts? You should only be generating a config once per project I imagine so to me it seems like a special case.

Comment (scalvert) [reply packages/cli/src/commands/config/init.ts]: But they're all code generators, no? I could see that argument, and I'm not super opinionated either way about this. We can always change it later if we want.

File: packages/core/src/configuration/config-init-service.ts (hunk 1 above)
Comment (mahirshah) [reply packages/core/src/configuration/config-init-service.ts]: good call, updated to use yeoman generators

File: packages/cli/src/commands/config/init.ts (hunk 3 above)
Comment (mahirshah) [reply packages/cli/src/commands/config/init.ts]: true, I also dont feel too strongly either way. I currently added under config:init, but feel free to move to under generate once the task/plugin generators are up.

**Comment by scalvert** [comment]
@mahirshah can you rebase with the latest changes in master? There's currently a conflict. There's a few open issues here still, in case you want to fix them up!

File: packages/cli/__tests__/generators/generate-config-test.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+describe('config-init-generator', () => {
+  it('should write a config given default answers', async () => {
+    const directory = await helpers.run(ConfigGenerator).withPrompts({
+      framework: 'Other',
+      format: CheckupConfigFormat.JSON,
+    });
+
+    assert.file('.checkuprc');
+    expect(fs.readFileSync(path.join(directory, '.checkuprc'), 'utf8')).toMatchSnapshot();
```
Comment (scalvert) [inline packages/cli/__tests__/generators/generate-config-test.ts]: I added a handy util for this stuff. Check out my generator tests.

`root.file('.checkuprc').contents` yada yada

File: packages/cli/__tests__/generators/generate-config-test.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  });
+
+  it('should write a config given Ember framework', async () => {
+    const directory = await helpers.run(ConfigGenerator).withPrompts({
+      framework: 'Ember',
+      format: CheckupConfigFormat.JSON,
+    });
+
+    assert.file('.checkuprc');
+    expect(fs.readFileSync(path.join(directory, '.checkuprc'), 'utf8')).toMatchSnapshot();
```
Comment (scalvert) [inline packages/cli/__tests__/generators/generate-config-test.ts]: Same as above.

File: packages/cli/__tests__/generators/generate-config-test.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  });
+
+  it('should error if a checkuprc file is already present', async () => {
+    await expect(
+      helpers.run(ConfigGenerator).inTmpDir(function(dir) {
+        fs.writeFileSync(path.join(dir, '.checkuprc'), JSON.stringify({}));
```
Comment (scalvert) [inline packages/cli/__tests__/generators/generate-config-test.ts]: Same as above.

File: packages/core/src/generators/config/init.ts [hunk 7]
```diff
@@ -0,0 +1,73 @@
+import * as Generator from 'yeoman-generator';
```
[the remark this answers is not in the document]
Comment (mahirshah) [reply packages/core/src/generators/config/init.ts]: Hmm I think you may be looking at an old diff? The generator is now in `packages/cli/src/generators/config.ts`

[the remark this answers is not in the document]
Comment (scalvert) [reply packages/core/src/generators/config/init.ts]: Yep this is an old comment.

File: packages/cli/__tests__/generators/generate-config-test.ts (hunk 4 above)
Comment (mahirshah) [reply packages/cli/__tests__/generators/generate-config-test.ts]: updated, thanks!

**Pull request merged**